chaos-mesh/chaos-mesh
# ci: add skywalking eyes license header check

GitHub Actions workflow to check for Apache 2.0 License in all applicable file headers. First run to fix headers automatically with some manual fixes for headers with incorrect dates or missing lines.



### What problem does this PR solve?


Issue Number: close #2237 

Problem Summary:

### What is changed and how it works?

What's Changed:

- Added an GitHub Actions workflow `.github/workflows/license-checker.yml`
- Added a configuration file for Apache Skywalking Eyes `.github/.licenserc.yaml`
- Approximately 700 files have been remediated with the correct License Header.

### Related changes

### Checklist


Tests



Side effects


### Release note 

```release-note
- Added CI step to check for Apache 2.0 License in applicable file headers.
- Updated existing License Headers, added to files with no License Header.
```

### Manual Tests

I have ran a number of linters to ensure no breaking code changes were made, along with spot checking around 30 files of various formats (.go, .ts, .tsx, .svg, .js, .yml, .toml) along with running the build and test `ci` steps successfully.
